### PR TITLE
Management API : allow empty body for creating tenant or device with generated ID

### PR DESCRIPTION
**Pull request by jbtrystram**
this fixes https://github.com/eclipse/hono/issues/1370

### CODE REVIEW

**Comment by sophokles73** [comment]
does this also allow omitting the content-type in the request?

**Comment by jbtrystram** [comment]
> does this also allow omitting the content-type in the request?

@sophokles73 i am not certain. 
`curl -i -X POST http://$REGISTRY_IP:28080/tenants ` this works fine, and nothing is specified. as there is no body i would expect no content type is specified but i don't master Curl enough to be certain.

**Comment by jbtrystram** [comment]
@sophokles73  `curl -i -H "content-type: " -X POST http://$REGISTRY_IP:28080/tenants `
this works as well. WDYT ?

**Comment by sophokles73** [comment]
@jbtrystram you should run `curl -v -X POST http://$REGISTRY_IP:28080/tenants` to see the headers of the request being sent. I assume that no *Content-Type* header will be sent ...

**Comment by jbtrystram** [comment]
> @jbtrystram you should run `curl -v -X POST http://$REGISTRY_IP:28080/tenants` to see the headers of the request being sent. I assume that no _Content-Type_ header will be sent ...

@sophokles73  indeed ! 
```
[hono] curl -v -X POST http://$REGISTRY_IP:28080/tenants                                                                                          getting-started  ✱
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 28080 (#0)
> POST /tenants HTTP/1.1
> Host: localhost:28080
> User-Agent: curl/7.64.0
> Accept: */*
> 
< HTTP/1.1 201 Created
< etag: c734d8a6-0172-4835-a37b-7a5237f840c1
< location: /tenants/64cc21b5-49ff-47a3-844f-6d0cb190a9f5
< content-type: application/json; charset=utf-8
< content-length: 45
< 
* Connection #0 to host localhost left intact
{"id":"64cc21b5-49ff-47a3-844f-6d0cb190a9f5"}% 
```

File: tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java [hunk 1]
```diff
@@ -179,15 +179,15 @@ public void testAddDeviceFailsForMissingContentType(final TestContext ctx) {
...
     /**
-     * Verifies that a device cannot be registered if the request
+     * Verifies that a device can be registered if the request
      * does not contain a body.
      * 
      * @param ctx The vert.x test context
      */
     @Test
-    public void testAddDeviceFailsForMissingBody(final TestContext ctx) {
+    public void testAddDeviceSucceedsForMissingBody(final TestContext ctx) {
```
Comment (sophokles73) [inline tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java]: I would suggest to name this *testAddDeviceSucceedsForEmptyBody* as the body is not missing in this case, is it?

Comment (jbtrystram) [reply tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java]: Indeed, in this case, the body seems to be an empty JsonObject.. Now I don't understand why the test fails, as it should work with an empty JsonObject, as demonstrated in my other PR..

**Comment by jbtrystram** [comment]
@sophokles73  I added a test to verify that it succeed with no body nor content-type set.

File: tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java (hunk 1 above)
Comment (sophokles73) [reply tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java]: from the perspective of the *registerDevice* method that is correct. However, we are not testing the client here but the service implementation and from the point of view of the service implementation the body is not *missing* but simply *empty*. The same holds true for the test that you have added ...

Comment (jbtrystram) [reply tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationHttpIT.java]: I updated the names

**Pull request merged**